tests/test_integration.py

```diff
@@ -2983,8 +2984,16 @@ def test_pip_issues_9420_workaround():
 
 def test_requirement_file_from_url(tmpdir):
     # type: (Any) -> None
+
+    constraints = os.path.join(str(tmpdir), "constraints.txt")
+    with open(constraints, "w") as fp:
+        fp.write("translate>=3.2.1,<3.6.0")
```

Comment:
Not a full pinning, but that's not trivial across 2.7 -> 3.9 CI test rigs and this works in the moment :/